lib/msf/ui/console/command_dispatcher/modules.rb

```diff
@@ -611,13 +607,18 @@ def cmd_search(*args)
               ::File.open(output_file, "wb") { |ofd|
                 ofd.write(tbl.to_csv)
               }
-            else
-              print_line(tbl.to_s)
-              print_module_search_results_usage
+              return true
+            end
+
+            print_line(tbl.to_s)
+            print_module_search_results_usage
 
-              print_status("Using #{used_module}") if used_module
+            if @module_search_results.length == 1 && use
+              used_module = @module_search_results_with_usage_metadata.first[:mod].fullname
+              cmd_use(used_module, true)
             end
 
+            print_status("Using #{used_module}") if used_module
```

Comment:
I think we talked about moving this above the `cmd_use` call - did we leave that out intentionally? 👀 

Current:

```
msf6 auxiliary(scanner/mysql/mysql_login) > use smb_login

Matching Modules
================

   #  Name                             Disclosure Date  Rank    Check  Description
   -  ----                             ---------------  ----    -----  -----------
   0  auxiliary/scanner/smb/smb_login  .                normal  No     SMB Login Check Scanner


Interact with a module by name or index. For example info 0, use 0 or use auxiliary/scanner/smb/smb_login

[*] New in Metasploit 6.4 - The CreateSession option within this module can open an interactive session
[*] Using auxiliary/scanner/smb/smb_login
```

Expected:

```
msf6 auxiliary(scanner/mysql/mysql_login) > use smb_login

Matching Modules
================

   #  Name                             Disclosure Date  Rank    Check  Description
   -  ----                             ---------------  ----    -----  -----------
   0  auxiliary/scanner/smb/smb_login  .                normal  No     SMB Login Check Scanner


Interact with a module by name or index. For example info 0, use 0 or use auxiliary/scanner/smb/smb_login

[*] Using auxiliary/scanner/smb/smb_login
[*] New in Metasploit 6.4 - The CreateSession option within this module can open an interactive session
```

Reply:
nope just forgot to move it